x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/TypeUtils.java

```diff
@@ -176,4 +177,22 @@ static EsType of(Class<? extends Object> clazz) throws SQLException {
         }
         return dataType;
     }
+
+    static EsType baseType(EsType type) throws SQLException {
+        String typeName = type.getName();
+        return isArray(type)
+            ? of(typeName.substring(0, typeName.length() - "_ARRAY".length()).toLowerCase(Locale.ROOT))
+            : type;
+    }
+
+    static EsType arrayOf(EsType type) throws SQLException {
```

Comment:
> It is because ENUM_NAME_TO_TYPE is local to TypeUtils?

Yes.

(`TypeUtils#isString()` is btw also only used once, in `JdbcPreparedStatement`, with no `TypeUtils` dependencies.)